lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -411,6 +906,250 @@ def upload_reads(self, ctx, params):
+
+        # Get the reads library
+        ws = workspaceService(self.workspaceURL, token=token)
+        ws_reads_ids = []
+        for read_name in params[self.PARAM_IN_LIB]:
+            ws_reads_ids.append({'ref': read_name})
+        try:
+            reads = ws.get_objects(ws_reads_ids)
+        except WorkspaceException as e:
+            self.log('Logging stacktrace from workspace exception:\n' + e.data)
+            raise
+
+        output = {}
+        for read_name, read in zip(params[self.PARAM_IN_LIB], reads):
+            self.log('=== processing read library ' + read_name + '===\n',
+                     prefix_newline=True)
+            output[read_name] = self.process_reads(
+                read, params[self.PARAM_IN_GZIP],
+                params[self.PARAM_IN_INTERLEAVED], token)
+        output = {'files': output}
+
+        #END convert_read_library_to_file
+
+        # At some point might do deeper type checking...
+        if not isinstance(output, dict):
+            raise ValueError('Method convert_read_library_to_file return value ' +
+                             'output is not type dict as required.')
+        # return the results
+        return [output]
+
+    def export_reads(self, ctx, params):
+        """
+        Using the convert_read_library_to_file function, this prepares a standard
+        KBase download package, zips it, and uploads to shock.
+        :param params: instance of type "ExportParams" -> structure:
+           parameter "input_ref" of String
+        :returns: instance of type "ExportOutput" -> structure: parameter
+           "shock_id" of String
+        """
+        # ctx is the context object
+        # return variables are: output
+        #BEGIN export_reads
+        print('running export_reads')
+        pprint(params)
+
+        # for some reason, in tests this doesn't happen, or something cleans out the shock
+        # temp directory.  So make sure the temp space exists.
+        self.shock_temp = os.path.join(self.scratch, self.SHOCK_TEMP)
+        self.mkdir_p(self.shock_temp)
+
+         # validate parameters
+        if 'input_ref' not in params:
+            raise ValueError('Cannot export reads- no input_ref field defined.')
+
+        # get WS metadata to get ws_name and obj_name
+        ws = workspaceService(url=self.workspaceURL)
+        info = ws.get_object_info_new({'objects':[{'ref': params['input_ref'] }],'includeMetadata':0, 'ignoreErrors':0})[0]
+
+        # export to a file, don't set any conversion parameters
+        read_libraries = [ params['input_ref'] ]
+        read_lib_info = self.convert_read_library_to_file(ctx, { 
+                              'read_libraries': read_libraries
+                            })[0]['files']
+        pprint(read_lib_info)
+
+        # create the output directory and move the file there
+        export_package_dir = os.path.join(self.scratch, info[1])
+        os.makedirs(export_package_dir)
+        for r in read_lib_info:
```

Comment:
this loop seems to be unnecessary since there's only one entry in read_lib_info and r = params['input_ref']